pkg/bootflow/types/state.go

```diff
@@ -0,0 +1,125 @@
+package types
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// State describes a virtual state of a machine being booted
+type State struct {
+	// Do not mutate these values from the outside, the mutability
+	// is owned by State itself.
+
+	SystemArtifacts SystemArtifacts
```

Comment:
Will we store somewhere the result of check of an artifact? Is it a TrustChain? Can TrustChain use each other?

Reply:
Ah sorry, seems like SystemArtifact is firmware, while pieces of it that we verify are in VerifiedData?

Reply:
> SystemArtifact is firmware, while pieces of it that we verify are in VerifiedData?

Yes. Though not only "firmware" might be `SystemArtifact`. For example "status registers" is also a `SystemArtifact`.